From 52a3fc78abd3f0ffab170e4c35beb93e58dee347 Mon Sep 17 00:00:00 2001
From: Sivashankari Madhavan <quic_sivamadh@quicinc.com>
Date: Fri, 19 Aug 2022 20:06:11 +0530
Subject: [PATCH] ath12k: Fix TPC CMD Info's sent as 0

Getting the reg power as -ive values in TPC stats.

As per the analysis, We have overwritten the NL80211_RRF_PSD flag by
NL80211_RRF_AUTO_BW.

Now changed this by OR the NL80211_RRF_AUTO_BW with flags

Signed-off-by: Sivashankari Madhavan <quic_sivamadh@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/reg.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/wireless/ath/ath12k/reg.c b/drivers/net/wireless/ath/ath12k/reg.c
index c410b2d..6e4d286 100644
--- a/drivers/net/wireless/ath/ath12k/reg.c
+++ b/drivers/net/wireless/ath/ath12k/reg.c
@@ -680,7 +680,7 @@ ath12k_reg_build_regd(struct ath12k_base *ab,
                                  k = 0;
                          }
                          reg_6g_number--;
-			flags = NL80211_RRF_AUTO_BW;
+			flags |= NL80211_RRF_AUTO_BW;
 		} else {
 			break;
 		}
-- 
2.17.1

